File: shopping-cart-microservices/orderService/src/main/java/com/microservice/orderService/Entity/Order.java
package com.microservice.orderService.Entity;

import jakarta.persistence.*;
import lombok.Data;

import java.util.Date;
import java.util.List;

@Data
@Entity
@Table(name = "orders")
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private Long customerId;

    @Column(nullable = false)
    private Date orderDate;

    @Column(nullable = false)
    private Double totalPrice;

    @Enumerated(EnumType.STRING)
    private OrderStatus status;

    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    @JoinColumn(name = "fk_oId" , referencedColumnName = "id")
    private List<OrderItem> orderItems;


}

File: shopping-cart-microservices-Front-end/E-Shop/src/app/model/Order.ts
import { OrderItems } from "./OrderItems";

export class Order{

    customerId!: number;
    totalPrice!: number;
    orderItems!: OrderItems [];

    constructor( customerId: number,totalPrice: number,orderItems:OrderItems []){
        this.customerId=customerId;
        this.totalPrice=totalPrice;
        this.orderItems=orderItems;
    }
}